src/nisarqa/products/product_reader.py

```diff
@@ -1500,7 +1553,12 @@ def _get_raster_from_path(
 
         # Get dataset object and check for correct dtype
         dataset = _get_dataset_handle(h5_file, raster_path)
-        kwargs["data"] = dataset
+
+        use_cache = True
```

Comment:
We'll make this an argument to the `NisarProduct` constructor. In addition, the user should be able to specify an optional `cache_dir` where the cached memory mapped files will be placed, defaulting to the [default temporary directory](https://docs.python.org/3/library/tempfile.html#tempfile.mkstemp).

(NOTE: I would call the directory `scratch_path` in the runconfig in order to match the name used in NISAR SAS runconfigs, but internally in the class constructor I think `cache_dir` is a more appropriate name.)

(NOTE NOTE: For some of the CalTools, we create a temporary JSON file. It would be good to put this in the same scratch directory, but I would probably name the parameter `scratch_dir` in that context, rather than `cache_dir`.)